src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -727,20 +728,31 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
                         // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        Vector128<sbyte> sourceValue;
+                        bool allBytesAreAscii;
 
                         // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
                         // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        if (Sse2.IsSupported)
+                        {
+                            sourceValue = Sse2.LoadVector128(startingAddress);
+                            allBytesAreAscii = Sse2.MoveMask(sourceValue) == 0;
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
+                            sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            allBytesAreAscii = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() >= 0;
+                        }
 
-                        if (index == 0)
+                        if (allBytesAreAscii)
                         {
                             // All of the following 16 bytes is ASCII.
 
                             if (Ssse3.IsSupported)
```

Comment:
>(Ssse3.IsSupported) [](start = 31, length = 19)

So are we not optimizing this path for `AdvSimd` or are we planning to do it in separate PR?

Reply:
I might not have enough time to work on that until the preview 8 snap. Depending on how it works out I might just add a TODO for future work.

Reply:
Sounds good.

---
In reply to: [453072386](https://github.com/dotnet/runtime/pull/38707#discussion_r453072386) [](ancestors = 453072386)